### PR TITLE
[Taxation] Add validation of negative tax rate

### PR DESCRIPTION
**Pull request by coldic3**
| Q               | A
| --------------- | -----
| Branch?         | master 
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/TaxationBundle/Resources/translations/validators.en.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ sylius:
             not_blank: Please enter tax rate name.
         amount:
             not_blank: Please enter tax rate amount.
+            invalid: The amount is invalid.
```
Comment (stloyd) [inline src/Sylius/Bundle/TaxationBundle/Resources/translations/validators.en.yml:16]:
```suggestion
            invalid: The tax rate amount is invalid.
```

File: src/Sylius/Bundle/TaxationBundle/Resources/translations/validators.en.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ sylius:
             not_blank: Please enter tax rate name.
         amount:
             not_blank: Please enter tax rate amount.
+            invalid: The tax rate amount is invalid.
```
Comment (Zales0123) [inline src/Sylius/Bundle/TaxationBundle/Resources/translations/validators.en.yml:16]: Alphabetical order R.I.P [*] 😢

File: src/Sylius/Bundle/TaxationBundle/Resources/translations/validators.pl.yml [hunk 3]
```diff
@@ -16,6 +16,7 @@ sylius:
             not_blank: Proszę podać nazwę stawki podatkowej.
         amount:
             not_blank: Proszę podać wartość stawki podatkowej.
+            invalid: Wartość stawki podatkowej jest nieprawidłowa.
```
Comment (AdamKasp) [inline src/Sylius/Bundle/TaxationBundle/Resources/translations/validators.pl.yml:19]: you should add translation only for English, the rest of it is translate by `crowdIn`

File: src/Sylius/Behat/Context/Ui/Admin/ManagingTaxRateContext.php [hunk 4]
```diff
@@ -238,6 +238,14 @@ public function iShouldBeNotifiedThatIsRequired($element)
         $this->assertFieldValidationMessage($element, sprintf('Please enter tax rate %s.', $element));
     }
 
+    /**
+     * @Then I should be notified that :element is invalid
+     */
+    public function iShouldBeNotifiedThatIsInvalid($element)
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingTaxRateContext.php:244]: Missing type hint and return type

File: src/Sylius/Bundle/TaxationBundle/Resources/translations/validators.pl.yml (hunk 3 above)
Comment (Zales0123) [inline src/Sylius/Bundle/TaxationBundle/Resources/translations/validators.pl.yml:19]: We should not change any non-English translations in the PR and let Crowdin to its job 🚀

File: src/Sylius/Bundle/TaxationBundle/Resources/translations/validators.en.yml (hunk 2 above)
Comment (coldic3) [reply src/Sylius/Bundle/TaxationBundle/Resources/translations/validators.en.yml:15]: fixed :slightly_smiling_face:

File: features/taxation/managing_tax_rates/tax_rate_validation.feature [hunk 5]
```diff
@@ -81,3 +81,12 @@ Feature: Tax rate validation
         When I remove its zone
         And I try to save my changes
         Then I should be notified that zone has to be selected
+
+    @ui
+    Scenario: Trying to add a new tax rate with negative amount
+        Given I want to create a new tax rate
+        When I name it "Food and Beverage Tax Rates"
```
Comment (GSadee) [inline features/taxation/managing_tax_rates/tax_rate_validation.feature:88]:
```suggestion
        When I want to create a new tax rate
        And I name it "Food and Beverage Tax Rates"
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "feature/tax-rate-without-negative-amount"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

**Review by SirDomin — APPROVED** [review]

**Review by Zales0123 — APPROVED** [review]

**Review by AdamKasp — APPROVED** [review]

File: features/taxation/managing_tax_rates/tax_rate_validation.feature (hunk 5 above)
Comment (coldic3) [reply features/taxation/managing_tax_rates/tax_rate_validation.feature:88]: It is consistent with other scenarios like [this one](https://github.com/Sylius/Sylius/blob/dc727c51ea51de39c7312769436f51610e12eb98/features/taxation/managing_tax_rates/tax_rate_unique_code_validation.feature#L15). Should I change it? If I change it, I will have to fix all scenarios with "Given I want to create a new tax rate".

Comment (coldic3) [reply features/taxation/managing_tax_rates/tax_rate_validation.feature]: I see now `Given` and `When` are the same thing for Behat so I've committed your suggestion. Thanks.

**Comment by GSadee** [comment]
Thank you, Kevin! :1st_place_medal:

**Comment by bashilbers** [comment]
Guys, why not use https://symfony.com/doc/current/reference/constraints/PositiveOrZero.html. In many cases 0% is also a perfectly fine tax percentage. This is now no longer possible because the `Positive` constraint does not allow 0.

**Pull request merged**